src/main/java/ch/njol/skript/util/SkriptColor.java

```diff
@@ -94,50 +125,42 @@ public ChatColor asChatColor() {
 	public DyeColor asDyeColor() {
 		return dye;
 	}
-	
+
 	@Override
 	public String getName() {
-		return name();
+		return Streams.findLast(names.entrySet().stream()
+				.filter(entry -> entry.getValue().equals(this))
+				.map(entry -> entry.getKey()))
+				.orElse(name);
 	}
-	
+
 	@Override
 	public String getFormattedChat() {
 		return "" + chat;
 	}
-	
+
 	@Deprecated
 	@Override
 	public byte getWoolData() {
 		return dye.getWoolData();
 	}
-	
+
 	@Deprecated
 	@Override
 	public byte getDyeData() {
 		return (byte) (15 - dye.getWoolData());
 	}
 	
-	final static Map<String, SkriptColor> names = new HashMap<>();
-	final static Set<SkriptColor> colors = new HashSet<>();
-	public final static String LANGUAGE_NODE = "colors";
-	
-	static {
-		for (SkriptColor color : values())
-			colors.add(color);
-		Language.addListener(new LanguageChangeListener() {
-			@Override
-			public void onLanguageChange() {
-				names.clear();
-				for (SkriptColor color : values()) {
-					String node = LANGUAGE_NODE + "." + color.name();
-					color.adjective = new Adjective(node + ".adjective");
-					for (String name : Language.getList(node + ".names"))
-						names.put(name.toLowerCase(), color);
-				}
-			}
-		});
+	public static SkriptColor[] values() {
+		return new SkriptColor[] {LIGHT_GREY, DARK_GREY, WHITE, BLACK, LIGHT_PURPLE, DARK_PURPLE, LIGHT_CYAN, DARK_BLUE, DARK_CYAN, BROWN, DARK_GREEN, LIGHT_GREEN, YELLOW, ORANGE, DARK_RED, LIGHT_RED};
 	}
 	
+	public static Optional<SkriptColor> valueOf(String name) {
+		return colors.stream()
```

Comment:
You could do `names.get(name.toLowerCase())` instead.

Reply:
That could work potentially wrapped in a nullable Optional.